### PR TITLE
Update github actions and added codecov badge

### PR DESCRIPTION
**Pull request by kmacdonald-stsci**

### CODE REVIEW

File: .github/workflows/spherical_geometry.yml [hunk 1]
```diff
@@ -73,3 +76,21 @@ jobs:
...
         file: ./coverage.xml
+
+  bandit:
+    name: Bandit
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python
+      uses: actions/setup-python@v1
```
Comment (pllim) [inline .github/workflows/spherical_geometry.yml:87]: Why v1 ?
```suggestion
      uses: actions/setup-python@v2
```

File: README.rst [hunk 2]
```diff
@@ -1,13 +1,17 @@
 User documentation
 ==================
 
+.. image:: https://codecov.io/gh/spacetelescope/spherical_geometry/branch/master/master/graphs/badge.svg
```
Comment (pllim) [inline README.rst:4]: Why not add the CI badge too, while you are at it?

**Review by pllim — COMMENTED** [review]
I don't think not all the jobs need to be run in scheduled cron. You can exclude the unnecessary ones using `if:` and filter by GitHub event.

File: .github/workflows/spherical_geometry.yml (hunk 1 above)
Comment (kmacdonald-stsci) [reply .github/workflows/spherical_geometry.yml:87]: I changed this.

File: README.rst (hunk 2 above)
Comment (kmacdonald-stsci) [reply README.rst:4]: I added a CI badge.

File: .github/workflows/spherical_geometry.yml [hunk 3]
```diff
@@ -73,3 +76,21 @@ jobs:
...
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.8
+    - name: Install Dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install bandit
+    - name: Security checks with bandit
+      run: |
+        bandit romancal -r -x tests
```
Comment (jdavies-st) [inline .github/workflows/spherical_geometry.yml:96]: `romancal` means this is not checking anything.

Comment (kmacdonald-stsci) [reply .github/workflows/spherical_geometry.yml]: Corrected this.

**Review by nden — APPROVED** [review]

**Pull request merged**